cli/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use clap::{App, AppSettings, Arg, ArgGroup, ArgMatches, SubCommand};
+use lazy_static::lazy_static;
+use maplit::hashmap;
+use rayon::prelude::IntoParallelIterator;
+use std::collections::{HashMap, VecDeque};
+use std::path::{Path, PathBuf};
+use std::process::{exit, Command};
+use std::sync::Mutex;
+use std::{env, fs};
+
+use log::{error, info, warn};
+
+// Usages:
+//  yew run directory/
+//  yew build --run directory/ (same as above)
+//  yew build directory/ (only builds)
+//  yew build examples/* (to build all examples)
+
+fn main() {
+    let matches = App::new("Yew CLI")
+        .version("0.1")
+        .about("Builds and runs Yew application projects")
+        .setting(AppSettings::SubcommandRequired)
+        .subcommand(
+            SubCommand::with_name("build")
+                .about("compiles a Yew application")
+                .arg(
+                    Arg::with_name("run")
+                        .help("Start a webserver for the built project and open it in a browser window")
+                        .long("run")
+                        .required(true)
+                        .short("r")
+                )
+                .arg(
+                    Arg::with_name("release")
```

Comment:
I think this could be better handled by taking everything after `--` and passing it to `cargo` (like how `cargo` allows you to pass arguments to `rustc`, e.g. `cargo test -- --nocapture` where `--nocapture` is passed to `rustc`).